src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1214 @@
+}
+
+enum Direction {
+    Right,
+    Left,
+    Up,
+    Down,
+    StartNewRow,
+    None,
+}
+
+enum Property {
+    None,
+    Number_Sequence,
+    Prime,
+    Negative,
+    Even,
+    Divisible_By_Three,
+    Divisible_By_Four,
+    Divisible_By_Five,
+    Divisible_By_Six,
+    Divisible_By_Seven,
+    Divisible_By_Eight,
+    Ends_With_One,
+    Ends_With_Two,
+    Ends_With_Three,
+    Ends_With_Four,
+    Ends_With_Five,
+    Ends_With_Six,
+    Ends_With_Seven,
+    Ends_With_Eight,
+    Ends_With_Nine,
+    Ends_With_Zero,
+    Triangular_Number,
+    Square_Number,
+    Pentagonal_Number,
+    Hexagonal_Number,
+    Heptagonal_Number,
+    Octagonal_Number,
+    Sum_Of_Two_Squares,
+    Fibonacci_Number,
+    Lucas_Number,
+    Abundant,
+    Perfect,
+    Deficient,
+    Semi_Prime,
+}
+
+enum PropertyVisualization {
+    None,
+    Color,
+    Circle,
+}
+
+enum Formula {
+    N,
+    N_SQUARED,
+}
+
+class VisualizerGrid extends VisualizerDefault {
+    name = 'Grid'
+    description = 'This puts numbers in a grid!'
+    amountOfNumbers = 40000
+    startingIndex = 0
+    showNumbers = false
+    message = ''
+    preset = Preset.Custom
+    pathType = PathType.Spiral
+    resetAndAugmentByOne = false
+    formula = Formula.N
+    backgroundColor = BLACK
+    numberColor = WHITE
+
+    //Path variables
+    x = 0
+    y = 0
+    currentDirection = Direction.Right
+    numberToTurnAt = 0
+    incrementForNumberToTurnAt = 1
+    whetherIncrementShouldIncrementForSpiral = true
+
+    property1 = Property.Prime
+    property1Visualization = PropertyVisualization.Color
+    property1MainColor = RED
+    property2 = Property.None
+    property2Visualization = PropertyVisualization.None
+    property2MainColor = BLUE
+    property3 = Property.None
+    property3Visualization = PropertyVisualization.None
+    property3MainColor = GREEN
+    property4 = Property.None
+    property4Visualization = PropertyVisualization.None
+    property4MainColor = YELLOW
+    property5 = Property.None
+    property5Visualization = PropertyVisualization.None
+    property5MainColor = ORANGE
+    property6 = Property.None
+    property6Visualization = PropertyVisualization.None
+    property6MainColor = PURPLE
+    property7 = Property.None
+    property7Visualization = PropertyVisualization.None
+    property7MainColor = CYAN
+    property8 = Property.None
+    property8Visualization = PropertyVisualization.None
+    property8MainColor = MAGENTA
+    property9 = Property.None
+    property9Visualization = PropertyVisualization.None
+    property9MainColor = VERDANT
+    property10 = Property.None
+    property10Visualization = PropertyVisualization.None
+    property10MainColor = VIOLET
+    property11 = Property.None
+    property11Visualization = PropertyVisualization.None
+    property11MainColor = MUSTARD
+    property12 = Property.None
+    property12Visualization = PropertyVisualization.None
+    property12MainColor = GRAY
+
+    params = {
```

Comment:
This object runs from line 193 to line 553 (360 lines). It's too big. Do you think we could create this object dynamically (i.e. in a loop) so that it doesn't take up so many lines?

Reply:
That sounds like a good idea. I do not know how to do this, but I will give it a shot or reach out for help.

Reply:
Well, I don't think you actually need the properties in an array. You just need to add them to your visualizer and to the params object in a loop. That loop should go in the constructor of your visualizer. The visualizer doesn't currently have a constructor, but you can add one -- be sure to call your super() constructor in it, first thing. Then the loop to add properties will have stuff like
```
for (let p = 1; p <= NumberOfProperties; ++p) {
  propertyName = 'Property' + p;
  propertyVisualizationName = <blah blah>;
  ... etc...
  this[propertyName] = <whatever the initial value of the pth Property should be>
  this[propertyVisualizationName] = <whatever the initial value of the pth Property Visualization should be>
  ... etc...
  this.params[propertyName] = getProperty(...correct arguments go here...)
  this.params[propertyVisualizationName] = getPropertyVisualization(...again, fill in the right arguments...)
  ... etc...
}
```

This has the advantage that you could for now set the NumberOfProperties to something that seems like enough (a half dozen, maybe? Have you ever wanted to visualize more than that at once?) and then it can be bumped up later if anyone ever actually needs more. Or if you already have a use case for 10 or 11, you can just set it to that, and then later it could still be bumped even higher with almost no work.